mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/binarystorage/BinaryStorageManager.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.mongodb.binarystorage;
+
+import java.lang.reflect.Field;
+import java.util.Collections;
+import java.util.EnumMap;
+import java.util.Map;
+
+import org.hibernate.ogm.datastore.mongodb.options.BinaryStorageType;
+import org.hibernate.ogm.datastore.mongodb.options.impl.BinaryStorageOption;
+import org.hibernate.ogm.datastore.mongodb.utils.TableEntityTypeMappingInfo;
+import org.hibernate.ogm.model.key.spi.EntityKey;
+import org.hibernate.ogm.model.spi.Tuple;
+import org.hibernate.ogm.options.spi.OptionsContext;
+import org.hibernate.ogm.options.spi.OptionsService;
+
+import com.mongodb.client.MongoDatabase;
+import org.bson.Document;
+
+/**
+ * @author Sergey Chernolyas &amp;sergey_chernolyas@gmail.com&amp;
+ */
+public class BinaryStorageManager {
+	private static final BinaryStorage NOOP_DELEGATOR = new NoopBinaryStore();
+
+	private final Map<BinaryStorageType, BinaryStorage> BINARY_STORAGE_MAP;
+
+	public BinaryStorageManager(MongoDatabase mongoDatabase) {
+		EnumMap<BinaryStorageType, BinaryStorage> map = new EnumMap<>( BinaryStorageType.class );
+		map.put( BinaryStorageType.GRID_FS, new GridFSBinaryStore( mongoDatabase ) );
+		BINARY_STORAGE_MAP = Collections.unmodifiableMap( map );
+	}
+
+	public  void storeContentToBinaryStorage(Document currentDocument, EntityKey entityKey, OptionsService optionService, Tuple tuple) {
+		Class entityClass = TableEntityTypeMappingInfo.getEntityClass( entityKey.getTable() );
```

Comment:
`TableEntityTypeMappingInfo` cannot be a static utility class, it needs to be bound to the factory one is working with.